[Building Error] Kernel 6.13 CachyOS

OS: CachyOS x86_64
Kernel: Linux 6.13.0-2-cachyos
Building Methods: Both AUR and Git clone

❯ tail /var/lib/dkms/bmi260/1.0.0/build/make.log
:3:92: error: expected string
3 | .section ".export_symbol","a" ; __export_symbol_bmi260_core_probe: ; .asciz "GPL" ; .asciz IIO_BMI260 ; .balign 8 ; .quad bmi260_core_probe ; .previous
| ^
3 errors generated.
make[3]: *** [/usr/lib/modules/6.13.0-2-cachyos/build/scripts/Makefile.build:194: bmi260_core.o] Error 1
make[2]: *** [/usr/lib/modules/6.13.0-2-cachyos/build/Makefile:1997: .] Error 2
make[1]: *** [/usr/lib/modules/6.13.0-2-cachyos/build/Makefile:251: __sub-make] Error 2
make[1]: Leaving directory "/var/lib/dkms/bmi260/1.0.0/build"
make: *** [Makefile:251: __sub-make] Error 2
make: Leaving directory "/usr/lib/modules/6.13.0-2-cachyos/build"

Also build failed on NixOS, but log is different

```log
these 2 derivations will be built:
  /nix/store/q7nrfawxr21nibvg5qb5zhdahdna3vx3-bmi260-1.0.0.drv
  /nix/store/0g6xzhdxaj2lmwv67a6z131w9715b13h-linux-zen-6.13.1-modules.drv
building '/nix/store/q7nrfawxr21nibvg5qb5zhdahdna3vx3-bmi260-1.0.0.drv'...
bmi260> Running phase: unpackPhase
bmi260> unpacking source archive /nix/store/idi8im8byv5wlkfjrcilsh4ncnqgd899-source
bmi260> source root is source
bmi260> Running phase: patchPhase
bmi260> Running phase: updateAutotoolsGnuConfigScriptsPhase
bmi260> Running phase: configurePhase
bmi260> no configure script, doing nothing
bmi260> Running phase: buildPhase
bmi260> build flags: SHELL=/nix/store/4fvc5fm8bszmkydng1ivrvr5cbvr1g60-bash-5.2p37/bin/bash KERNEL_SRC=/nix/store/dnvwl2gl7mh65nzbxyv4q1pik0yyyg4q-linux-zen-6.13.1-dev/lib/modules/6.13.1-zen1/build
bmi260> make -C /nix/store/dnvwl2gl7mh65nzbxyv4q1pik0yyyg4q-linux-zen-6.13.1-dev/lib/modules/6.13.1-zen1/build M=/build/source modules
bmi260> make[1]: Entering directory '/nix/store/dnvwl2gl7mh65nzbxyv4q1pik0yyyg4q-linux-zen-6.13.1-dev/lib/modules/6.13.1-zen1/build'
bmi260> make[2]: Entering directory '/build/source'
bmi260>   CC [M]  bmi260_core.o
bmi260> /build/ccO5xj4O.s: Assembler messages:
bmi260> /build/ccO5xj4O.s:7: Error: junk at end of line, first unrecognized character is `I'
bmi260> /build/ccO5xj4O.s:8: Error: junk at end of line, first unrecognized character is `I'
bmi260> /build/ccO5xj4O.s:9: Error: junk at end of line, first unrecognized character is `I'
bmi260> make[4]: *** [/nix/store/dnvwl2gl7mh65nzbxyv4q1pik0yyyg4q-linux-zen-6.13.1-dev/lib/modules/6.13.1-zen1/source/scripts/Makefile.build:194: bmi260_core.o] Error 1
bmi260> make[3]: *** [/nix/store/dnvwl2gl7mh65nzbxyv4q1pik0yyyg4q-linux-zen-6.13.1-dev/lib/modules/6.13.1-zen1/source/Makefile:1987: .] Error 2
bmi260> make[2]: *** [/nix/store/dnvwl2gl7mh65nzbxyv4q1pik0yyyg4q-linux-zen-6.13.1-dev/lib/modules/6.13.1-zen1/source/Makefile:251: __sub-make] Error 2
bmi260> make[2]: Leaving directory '/build/source'
bmi260> make[1]: *** [/nix/store/dnvwl2gl7mh65nzbxyv4q1pik0yyyg4q-linux-zen-6.13.1-dev/lib/modules/6.13.1-zen1/source/Makefile:251: __sub-make] Error 2
bmi260> make[1]: Leaving directory '/nix/store/dnvwl2gl7mh65nzbxyv4q1pik0yyyg4q-linux-zen-6.13.1-dev/lib/modules/6.13.1-zen1/build'
bmi260> make: *** [Makefile:15: modules] Error 2
```

--- a/bmi260.h
+++ b/bmi260.h
@@ -2,6 +2,7 @@
 #ifndef BMI260_H_
 #define BMI260_H_
 
+#include <linux/version.h>
 #include <linux/iio/iio.h>
 #include <linux/regulator/consumer.h>
 
@@ -34,5 +35,14 @@ int bmi260_core_probe(struct device *dev, struct regmap *regmap,
 int bmi260_enable_irq(struct regmap *regmap, enum bmi260_int_pin pin, bool enable);
 
 int bmi260_probe_trigger(struct iio_dev *indio_dev, int irq, u32 irq_type);
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 13, 0)
+#define BMI260_EXPORT_SYMBOL(symbol)	EXPORT_SYMBOL_NS_GPL(symbol, "IIO_BMI260");
+#define BMI260_IMPORT_NS		MODULE_IMPORT_NS("IIO_BMI260");
+#else
+#define BMI260_EXPORT_SYMBOL(symbol)	EXPORT_SYMBOL_NS_GPL(symbol, IIO_BMI260);
+#define BMI260_IMPORT_NS		MODULE_IMPORT_NS(IIO_BMI260);
+#endif
+
 
 #endif  /* BMI260_H_ */

--- a/bmi260_core.c
+++ b/bmi260_core.c
@@ -141,7 +141,7 @@ const struct regmap_config bmi260_regmap_config = {
 	.reg_bits = 8,
 	.val_bits = 8,
 };
-EXPORT_SYMBOL_NS(bmi260_regmap_config, IIO_BMI260);
+BMI260_EXPORT_SYMBOL(bmi260_regmap_config);
 
 struct bmi260_regs {
 	u8 data; /* LSB byte register for X-axis */
@@ -680,7 +680,7 @@ int bmi260_enable_irq(struct regmap *regmap, enum bmi260_int_pin pin, bool enabl
 				     mask, enable_bit,
 				     BMI260_NORMAL_WRITE_USLEEP);
 }
-EXPORT_SYMBOL_NS(bmi260_enable_irq, IIO_BMI260);
+BMI260_EXPORT_SYMBOL(bmi260_enable_irq);
 
 static int bmi260_get_irq(struct fwnode_handle *fwnode, enum bmi260_int_pin *pin)
 {
@@ -979,7 +979,7 @@ int bmi260_core_probe(struct device *dev, struct regmap *regmap,
 
 	return devm_iio_device_register(dev, indio_dev);
 }
-EXPORT_SYMBOL_NS_GPL(bmi260_core_probe, IIO_BMI260);
+BMI260_EXPORT_SYMBOL(bmi260_core_probe);
 
 MODULE_AUTHOR("Justin Weiss <justin@justinweiss.com>");
 MODULE_DESCRIPTION("Bosch BMI260 driver");

--- a/bmi260_i2c.c
+++ b/bmi260_i2c.c
@@ -74,4 +74,4 @@ module_i2c_driver(bmi260_i2c_driver);
 MODULE_AUTHOR("Justin Weiss <justin@justinweiss.com>");
 MODULE_DESCRIPTION("BMI260 I2C driver");
 MODULE_LICENSE("GPL v2");
-MODULE_IMPORT_NS(IIO_BMI260);
+BMI260_IMPORT_NS;

